File: hw/chavez/u3/hw22SOLIDMaze/hw22Java/MazeSystem/src/main/java/ec/edu/espe/mazesystem/model/Maze.java
package ec.edu.espe.mazesystem.model;

import ec.edu.espe.mazesystem.generator.MazeGeneratorAlgorithm;
import ec.edu.espe.mazesystem.generator.MazeGenerator;

public class Maze extends AbstractMaze {
    private MazeGeneratorAlgorithm generator;

    public Maze(int width, int height, MazeGeneratorAlgorithm generator) {
        super(width, height);  
        this.generator = generator;
    }

    public Maze(int width, int height) {
        super(width, height);
        this.generator = new MazeGenerator();  
    }

    public Maze() {
        super(5, 5);
        this.generator = new MazeGenerator();
    }

    @Override
    public void generate() {
        if (generator != null) {
            generator.generateMaze(this);
        }
    }

    public void setGenerator(MazeGeneratorAlgorithm generator) {
        this.generator = generator;
    }

    public MazeGeneratorAlgorithm getGenerator() {
        return generator;
    }
}

File: hw/tipantiza/u3/hw22MazeSolidPrinciples/MazeSystem_py/model/maze.py
from model.abstract_maze import AbstractMaze
from generator.maze_generator import MazeGenerator


class Maze(AbstractMaze):

    def __init__(self, width=5, height=5, generator=None):
        super().__init__(width, height)

        if generator is None:
            generator = MazeGenerator()

        self.generator = generator

    def generate(self):
        if self.generator is not None:
            self.generator.generate_maze(self)

    def set_generator(self, generator):
        self.generator = generator

    def get_generator(self):
        return self.generator